src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -92,7 +94,15 @@ public void StopFileChangeListening()
             // and even if that happens, it will be just a perf hit
             if (_fileChangeCookie != s_none)
             {
-                Marshal.ThrowExceptionForHR(_fileChangeService.UnadviseFileChange(_fileChangeCookie.Value));
+                var hr = _fileChangeService.UnadviseFileChange(_fileChangeCookie.Value);
+
+                // Report non-fatal watson for failure.
+                if (hr != VSConstants.S_OK && File.Exists(_filePath))
```

Comment:
can it return S_FALSE? asking since I dont know what HR Marshal.ThrowExceptionForHR is actually checking. and what UnadviseFileChange return.